contrib/enfuse_pro_2.lua

```diff
@@ -0,0 +1,1050 @@
+        sldr_exposure_weight.value=dt.preferences.read("enfuse_pro", pref_text.."exposure_weight", "float")
+        sldr_saturation_weight.value=dt.preferences.read("enfuse_pro", pref_text.."saturation_weight", "float")
+        sldr_contrast_weight.value=dt.preferences.read("enfuse_pro", pref_text.."contrast_weight", "float")
+        sldr_exposure_optimum_weight.value=dt.preferences.read("enfuse_pro", pref_text.."exposure_optimum_weight", "float")
+        sldr_exposure_width_weight.value=dt.preferences.read("enfuse_pro", pref_text.."exposure_width_weight", "float")   
+	
+    end,
+    "1 - DRI image", "2 - DRI image","3 - DRI image","1 - DFF image", "2 - DFF image","3 - DFF image","free preset",  
+    
+    reset_callback = function(self_type)
+       cmbx_fusion_type.value = dt.preferences.read("enfuse_pro", "def_fusion_type", "integer")
+    end
+} 
+entry_tag = dt.new_widget("entry"){
+	tooltip = "Additional tags to be added on import. Seperate with commas, all spaces will be removed",
+	text = pref_addtags,
+	placeholder = "Enter tags, seperated by commas",
+	editable = true
+}
+local widget = dt.new_widget("box") {
+    orientation = "vertical",
+    label_align_options,
+    chkbtn_radial_distortion,
+    chkbtn_optimize_field,
+    chkbtn_optimize_image_center,
+    chkbtn_auto_crop,
+    chkbtn_distortion,
+    cmbx_grid_size,
+    cmbx_control_points,
+    cmbx_control_points_remove,
+    cmbx_correlation,
+    separator1,
+    separator1b,
+    label_enfuse_options,
+    cmbx_fusion_type,
+    chkbtn_image_variations,
+    sldr_exposure_weight,
+    sldr_saturation_weight,
+    sldr_contrast_weight,
+    sldr_exposure_optimum_weight,
+    sldr_exposure_width_weight, 
+    chkbtn_hard_masks,
+    chkbtn_save_masks,
+    cmbx_contrast_edge_scale,
+    cmbx_contrast_min_curvature,
+    cmbx_contrast_window_size,
+    button_save_preset,
+    button_load_preset,
+    separator2,
+    separator2b,
+    label_output_format,
+    cmbx_output_format,
+    label_path,
+    file_chooser_button_path,
+	chkbtn_source_location,
+    cmbx_existing_file,
+	cmbx_style,
+	chkbtn_cpytags,
+	entry_tag,
+}
+
+-- FUNCTION -- 
+local function GetFileName(full_path)
+	--[[Parses a full path (path/filename_identifier.extension) into individual parts
+	Input: Folder1/Folder2/Folder3/Img_0001.CR2
+	
+	Returns:
+	path: Folder1/Folder2/Folder3/
+	filename: Img_0001
+	identifier: 0001
+	extension: .CR2
+	
+	EX:
+	path_1, file_1, id_1, ext_1 = GetFileName(full_path_1)
+	]]
+	local path = string.match(full_path, ".*[\\/]")
+	local filename = string.gsub(string.match(full_path, "[%w-_]*%.") , "%." , "" ) 
+	local identifier = string.match(filename, "%d*$")
+	local extension = string.match(full_path, "%.%w*")
+    return path, filename, identifier, extension
+end
+local function truncate(x)
+      return x<0 and math.ceil(x) or math.floor(x)
+end 
+local function replace_comma_to_dot(s)
+	return string.gsub(s, "%,", ".")
+end
+local function file_exists(name)
```

Comment:
You could just df.check_if_file_exists() and save yourself some code

Reply:
I just replaced the only location the file_exists() function was called with df.create_unique_filename, so this has been removed entirely.